### PR TITLE
Custom wrapper to handle commands in oC

### PR DESCRIPTION
**Pull request by sharidas**
A custom wrapper to handle command transfer-ownership.
In this wrapper we can make changes for functions like
fopen or any other routines.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

A custom wrapper created along with a custom view so that changes required for common functions like fopen arguments or so can be handled here. Primarily the focus is for command transfer-ownership and later we would use for other command like recreate master key.

## Motivation and Context

This change would prevent the modification to common routines which were made to arguments of functions like fopen or copy etc. And use this wrapper for custom changes.

## How Has This Been Tested?

WIP

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/Files/CustomView.php [hunk 1]
```diff
@@ -0,0 +1,2065 @@
...
+use OC\Files\Cache;
+use OC\Files\Filesystem;
+use OC\Files\Storage\Storage;
+use OC\Memcache\ArrayCache;
+use OCP\Files\Mount\IMountPoint;
+use OCP\Lock\ILockingProvider;
+use OC\Files\Mount\MoveableMount;
+use OCP\Lock\LockedException;
+
+class CustomView {
```
Comment (PVince81) [inline lib/private/Files/CustomView.php]: why ? and if really needed, why not `extend View` ?

Comment (sharidas) [inline lib/private/Files/CustomView.php]: Modifying to go with ```extend View```

File: lib/private/Files/CustomView.php [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+	/**
+	 * Rename/move a file or folder from the source path to target path.
+	 *
+	 * @param string $path1 source path
+	 * @param string $path2 target path
+	 *
+	 * @return bool|mixed
+	 */
+	public function renameCustom($path1, $path2) {
```
Comment (PVince81) [inline lib/private/Files/CustomView.php]: mind posting a diff between rename() and renameCustom() ? they look very similar and I don't really see what the difference is and why this is needed

Comment (sharidas) [inline lib/private/Files/CustomView.php]: Here is the diff between `rename()` and `renameCustom()`

```
sujith@sujith-Inspiron-5567 ~ $ diff rename_function.php customrename_function.php
9c9
<       public function rename($path1, $path2) {
---
>       public function renameCustom($path1, $path2) {
80d79
<                                       // moving a file/folder between storages (from $storage1 to $storage2)
82c81
<                                       $result = $storage2->moveFromStorage($storage1, $internalPath1, $internalPath2);
---
>                                       $result = $storage2->moveFromStorageCustom($storage1, $internalPath1, $internalPath2, true, true);
sujith@sujith-Inspiron-5567 ~ $ 
```
The difference is the call to moveFromStorage in customView's renameCustom() method. Rest of the code is same.

**Comment by PVince81** [comment]
Thanks. As discussed, try getting rid of the `CustomView` as it causes too much code duplication.

Since this custom view is calling a storage method with extra args, try simply writing that method call `moveFromStorage` and `copyFromStorage` directly inside the CustomEncryptionWrapper, and then set the two flags when passing down the call.

In general I'm still not happy that we need to add these extra flags to a "standard" storage method, we'll need to think of a better solution. Maybe there's a way to move the logic that enables itself through these flags, move that logic to the CustomEncryptionWrapper.

**Comment by PVince81** [comment]
@sharidas obsolete ? please close and link to your new PR

**Comment by sharidas** [comment]
Obsolete PR. The PR which implements the design or implementation for the change is https://github.com/owncloud/core/pull/28774

**Pull request closed without merging**